src/nvim/shada.c

```diff
@@ -2174,6 +2174,11 @@ static inline ShaDaWriteResult shada_read_when_writing(ShaDaReadDef *const sd_re
           shada_free_shada_entry(&entry);
           break;
         }
+        if (wms->global_marks[idx].data.type == kSDItemMissing) {
+          if (namedfm[idx].fmark.timestamp >= entry.timestamp) {
+            break;
```

Comment:
I think because you're discarding this entry, it has to be freed here?
```suggestion
            shada_free_shada_entry(&entry);
            break;
```

Reply:
Yes, this is what I was thinking as well. I'll try it out and see.